api/src/test/java/org/openmrs/test/StartModuleExecutionListener.java

```diff
@@ -89,12 +90,16 @@ public void prepareTestInstance(TestContext testContext) throws Exception {
 				Properties props = BaseContextSensitiveTest.runtimeProperties;
 				props.setProperty(ModuleConstants.RUNTIMEPROPERTY_MODULE_LIST_TO_LOAD, modulesToLoad);
 				try {
+					Context.addProxyPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES);
```

Comment:
I do not like this. It is too permissive. Why do we need to add the proxy privilege for the entire module startup process?

Reply:
Without adding this we get test errors similar to this:

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.096 s <<< FAILURE! -- in org.openmrs.annotation.OpenmrsProfileExcludeFilterWithModulesJUnit4Test
[ERROR] org.openmrs.annotation.OpenmrsProfileExcludeFilterWithModulesJUnit4Test.match_shouldNotIncludeBeanIfModuleIsStarted -- Time elapsed: 0 s <<< ERROR!
java.lang.RuntimeException: Database out of sync with code: Anonymous role does not exist
	at org.openmrs.api.context.UserContext.getAnonymousRole(UserContext.java:370)
	at org.openmrs.api.context.UserContext.hasPrivilege(UserContext.java:347)
	at org.openmrs.api.context.Context.hasPrivilege(Context.java:736)
	at org.openmrs.aop.AuthorizationAdvice.before(AuthorizationAdvice.java:73)
	at org.springframework.aop.framework.adapter.MethodBeforeAdviceInterceptor.invoke(MethodBeforeAdviceInterceptor.java:57)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:241)
	at com.sun.proxy.$Proxy181.getGlobalProperty(Unknown Source)
	at org.openmrs.module.ModuleFactory.getModulesThatShouldStart(ModuleFactory.java:291)
	at org.openmrs.module.ModuleFactory.startModules(ModuleFactory.java:236)
	at org.openmrs.module.ModuleUtil.startup(ModuleUtil.java:130)
	at org.openmrs.test.StartModuleExecutionListener.prepareTestInstance(StartModuleExecutionListener.java:94)
	at org.springframework.test.context.TestContextManager.prepareTestInstance(TestContextManager.java:248)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.createTest(SpringJUnit4ClassRunner.java:227)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner$1.runReflectiveCall(SpringJUnit4ClassRunner.java:289)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.methodBlock(SpringJUnit4ClassRunner.java:291)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:246)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:190)
```

Module startup needs access to global properties when getting the module repository:

https://github.com/openmrs/openmrs-core/blob/e962f22be8ec4f02998e0f13b7918b792de71dcf/api/src/main/java/org/openmrs/module/ModuleUtil.java#L481-L485

Reply:
This still does not mean that we give the privilege to the entire module startup. Not so? :)

Reply:
Even then, we could mask bugs. We need to locate the specific places that need them.

Reply:
I don't think we should add privileges within the ModuleFactory as it is not a test class.